base/pkg/entry.jl

```diff
@@ -51,7 +51,7 @@ function add(pkg::AbstractString, vers::VersionSet)
     @sync begin
         @async if !edit(Reqs.add,pkg,vers)
             ispath(pkg) || throw(PkgError("unknown package $pkg"))
-            info("Nothing to be done")
+            info("`$pkg` package is already installed")
```

Comment:
Skimming the code and thinking my own empirical experience with when I see the 'nothing to be done' message makes me thing that doesn't happen, but I honestly can't be 100% sure.